Bloom.md

```diff
@@ -0,0 +1,341 @@
+operations (BF.ADD/BF.MADD/BF.INSERT) to continue without being rejected even when a filter is at full capacity. This will
+increase the false positve error rate, but a user can opt into this behavior to allow add operations to "succeed".
+
+### Scalable filters
+
+Bloom Filters can either be configured as scalable (default) or non-scalable through specification with the BF.RESERVE or BF.INSERT command.
+
+When scaling filters reach their capacity, if a user adds an item to the bloom object, a new bloom filter is created and
+added to the list of bloom filters within the same bloom object. This new bloom filter will have a larger capacity
+(previous bloom filter's capacity * expansion rate of the bloom object).
+
+When we want to check whether an item exists on a bloom object (BF.EXISTS/BF.MEXISTS), we look through each filter
+(from oldest to newest) in the object's filter list and perform a check operation on each one. Similarly, to add a new
+item to the bloom object, we check through all the filters to see if the item already exists and if not, the item is
+added to the current filter.
+
+When a bloom object has a larger number of bloom filters, it will result in reduced performance.
+
+Additionally, the default expansion rate is 2 and auto scaling of filters is enabled by default. The table below shows
+the total capacity (across all filters) when there are x additional (scaled out) filters and the starting filter has a
+capacity of 1.
+
+| x   | Capacity with x additional filters    |
+|-----|---------------------------------------|
+| 0   | 1                                     |
+| 1   | 3                                     |
+| 2   | 7                                     |
+| 3   | 15                                    |
+| 4   | 31                                    |
+| 5   | 63                                    |
+| 10  | 2047                                  |
+| 15  | 65535                                 |
+| 20  | 2097151                               |
+| 25  | 67108863                              |
+| 30  | 2147483647                            |
+| 35  | 68719476735                           |
+| 40  | 2199023255551                         |
+| 45  | 70368744177663                        |
+| 50  | 2251799813685247                      |
+| 55  | 72057594037927934                     |
+| 60  | 2305843008139952126                   |
+| 65  | 73786976294838205662                  |
+| 70  | 2361183241434822608686                |
+| 75  | 75557862452714323477986               |
+| 100 | 2535301200456458804968143894986       |
+
+Practically, we can expect scaling to stop well before the 50 range. With the default capacity of 100,000 and the default
+expansion rate of 2, the total items inserted (across all filters) would exceed the 64-bit unsigned integer limit before
+reaching 50 filters.
+
+
+### Choice of Bloom Filter Library
```

Comment:
Is the memory for this implementation managed by the same memory allocator as Redis? Or is it separate?